test/calibration/test_calibrations.py

```diff
@@ -0,0 +1,374 @@
+
+        xp0 = self.cals.get_schedule("xp", (0,))
+        xp3 = self.cals.get_schedule("xp", (3,))
+
+        self.assertEqual(xp0.instructions[0][1].pulse.sigma, 50)
+        self.assertEqual(xp3.instructions[0][1].pulse.sigma, 50)
+
+        # Check that we have the expected parameters in the calibrations.
+        expected = {self.amp_xp, self.amp, self.sigma, self.beta}
+        self.assertEqual(len(set(self.cals.parameters.keys())), len(expected))
+
+    def test_parameter_filtering(self):
+        """Test that we can properly filter parameter values."""
+
+        self._add_parameters()
+
+        # Check that these values are split between the qubits.
+        amp_values = self.cals.parameters_table(parameters=["amp"], qubit_list=[(0,)])
+        self.assertEqual(len(amp_values), 1)
+
+        # Check that we have one value for sigma.
+        sigma_values = self.cals.parameters_table(parameters=["σ"])
+        self.assertEqual(len(sigma_values), 1)
+
+        # Check that we have two values for amp.
+        amp_values = self.cals.parameters_table(parameters=["amp"])
+        self.assertEqual(len(amp_values), 2)
+
+        amp_values = self.cals.parameters_table(parameters=["amp"], qubit_list=[(3,)])
+        self.assertEqual(len(amp_values), 1)
+
+        # Check to see if we get back the two qubits when explicitly specifying them.
+        amp_values = self.cals.parameters_table(parameters=["amp"], qubit_list=[(3,), (0,)])
+        self.assertEqual(len(amp_values), 2)
+
+
+class TestControlChannels(QiskitTestCase):
+    """Test more complex schedules such as an echoed cross-resonance."""
+
+    def setUp(self):
+        """Create the setup we will deal with."""
+        controls = {(3, 2): [ControlChannel(10), ControlChannel(123)],
+                    (2, 3): [ControlChannel(15), ControlChannel(23)]}
+        self.cals = Calibrations(control_config=controls)
+
+        self.amp_cr = Parameter("amp_cr")
+        self.amp_rot = Parameter("amp_rot")
+        self.amp = Parameter("amp")
+        self.d0_ = DriveChannel(Parameter('ch0'))
+        self.d1_ = DriveChannel(Parameter('ch1'))
+        self.c1_ = ControlChannel(Parameter('ch0.1'))
+        self.sigma = Parameter("σ")
+        self.width = Parameter("w")
+        self.date_time = datetime.strptime("15/09/19 10:21:35", "%d/%m/%y %H:%M:%S")
+
+        cr_tone = GaussianSquare(640, self.amp_cr, self.sigma, self.width)
+        rotary = GaussianSquare(640, self.amp_rot, self.sigma, self.width)
+
+        with pulse.build(name="xp") as xp:
+            pulse.play(Gaussian(160, self.amp, self.sigma), self.d0_)
+
+        with pulse.build(name="cr") as cr:
+            with pulse.align_sequential():
+                with pulse.align_left():
+                    pulse.play(rotary, self.d1_)  # Rotary tone
+                    pulse.play(cr_tone, self.c1_)  # CR tone.
+                with pulse.align_sequential():
```

Comment:
Leaving out the `align_sequential` will not work. The reason is that the `cr` and `rotary` tones are played on channels `d1_` and `c1_` respectively. The echo is played on channel `d0_` you therefore need the `align_sequential` otherwise everything will overlap as each pulse is on a different channel.

Reply:
Hmm, I would have thought the outer `align_sequential` would take care of that. I mean that I would have thought these two were equivalent:

```
            with pulse.align_sequential():
                with pulse.align_left():
                    pulse.play(rotary, self.d1_)  # Rotary tone
                    pulse.play(cr_tone, self.c1_)  # CR tone.
                with pulse.align_sequential():
                    pulse.call(xp)
```

and

```
            with pulse.align_sequential():
                with pulse.align_left():
                    pulse.play(rotary, self.d1_)  # Rotary tone
                    pulse.play(cr_tone, self.c1_)  # CR tone.
                pulse.call(xp)
```

Reply:
Ah of course! I get the point. See here: b4a0f74

Reply:
Thanks, it was a bit of nit-picky comment but I was mainly asking just to check I wasn't misunderstanding something about the builder interface.

Reply:
Your understanding was correct. This also lead to a few lines of code less which is always good.